lib/diagnostic.js

```diff
@@ -5,17 +5,26 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
-
+const min = function (workingArray, callback) {
+  const newArray = [];
+  workingArray.reduce( function(element, index, array) {
```

Comment:
Since the function is assumed to be called with numbers as arguments, this will throw an error because you're trying to invoke ```reduce```, which is an array method, on a number.